cs/src/Contracts/TunnelPort.cs

```diff
@@ -4,109 +4,124 @@
-        /// </summary>
-        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-        public TunnelOptions? Options { get; set; }
-
-        /// <summary>
-        /// Gets or sets current connection status of the tunnel port.
-        /// </summary>
-        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-        public TunnelPortStatus? Status { get; set; }
-
-        /// <summary>
-        /// Gets or sets the username for the ssh service user is trying to forward.
-        /// </summary>
-        /// <remarks>
-        /// Should be provided if the <see cref="TunnelProtocol"/> is Ssh.
-        /// </remarks>
-        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-        public string? SshUser { get; set; }
     }
+
+    /// <summary>
+    /// Gets or sets the ID of the cluster the tunnel was created in.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [RegularExpression(ClusterIdPattern)]
+    [StringLength(ClusterIdMaxLength, MinimumLength = ClusterIdMinLength)]
+    public string? ClusterId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the generated ID of the tunnel, unique within the cluster.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [RegularExpression(TunnelIdPattern)]
+    [StringLength(TunnelIdLength, MinimumLength = TunnelIdLength)]
+    public string? TunnelId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the IP port number of the tunnel port.
+    /// </summary>
+    public ushort PortNumber { get; set; }
+
+    /// <summary>
+    /// Gets or sets the optional short name of the port.
+    /// </summary>
+    /// <remarks>
+    /// The name must be unique among named ports of the same tunnel.
+    /// </remarks>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [RegularExpression(TunnelNamePattern)]
+    [StringLength(TunnelNameMaxLength)]
+    public string? Name { get; set; }
+
+    /// <summary>
+    /// Gets or sets the optional description of the port.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [StringLength(DescriptionMaxLength)]
+    public string? Description { get; set; }
+
+    /// <summary>
+    /// Gets or sets the tags of the port.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [MaxLength(MaxTags)]
+    [ArrayStringLength(TagMaxLength, MinimumLength = TagMinLength)]
+    [ArrayRegularExpression(TagPattern)]
+    public string[]? Tags { get; set; }
+
+    /// <summary>
+    /// Gets or sets the protocol of the tunnel port.
+    /// </summary>
+    /// <remarks>
+    /// Should be one of the string constants from <see cref="TunnelProtocol"/>.
+    /// </remarks>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [StringLength(TunnelProtocol.MaxLength)]
+    public string? Protocol { get; set; }
```

Comment:
Add a regex validation with all know valid protocols

Reply:
How much value would there be in that, really? Can a fuzzer even generate values that match a regex?

It could be cleaner to write a custom validation attribute that checks if the value matches a list of strings. But the service already effectively does that anyway, so I think that would not be very useful either.